lib/cdo/user_helpers.rb

```diff
@@ -20,7 +20,7 @@ def self.generate_username(queryable, name)
     return prefix if queryable.where(username: prefix).limit(1).empty?
 
     # Throw darts to find an appropriate suffix, using it if we hit bullseye.
-    (0..1).each do |exponent|
+    (0..2).each do |exponent|
       min_index = 10**exponent
       max_index = 10**(exponent + 1) - 1
       3.times do |_i|
```

Comment:
I think your change doesn't actually increase the number of darts thrown. It increases the target space and reduces likelihood of collision. The increased target space is a fine thing. But in addition, how about increasing the number of tries from 3 to something like 20? I bet we'd see very few fallbacks to the slow query.